rosetta-source/src/main/rosetta/event-position-type.rosetta

```diff
@@ -42,3 +48,35 @@ type Portfolio: <" A Portfolio represents an aggregation of multiple Positions,
 
 	aggregationParameters AggregationParameters (1..1) <"Describes the portfolio by describing how to aggregate all its relevant Events.">
 	portfolioState PortfolioState (1..1) <"Describes the state of the Portfolio as a list of Positions resulting from the aggregation.">
+
+type InventoryBase: <"A data type that can be used to describe an inventory of securities.">
+    inventoryRecord InventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+
+type InventoryRecordBase: <"An individual piece of inventory. This represents a single security.">
+    identifer AssignedIdentifier (1..1) <"Unique identifier for this record. This can be used to uniquely identify a specific piece of inventory.">
+    security Security (1..1) <"The security details.">
+
+type Inventory extends InventoryBase: <"Describes an inventory of securities.">
+
+type InventoryRecord extends InventoryRecordBase: <"Describes an individual piece of inventory.">
+
+type AvailableInventory: <"A data type that can be used to describe the inventory of securities that a party holds. ">
+    party Party (0..*) <"Defines all parties involved for the list of inventory records in this set of inventory. For example, when used to describe securities lending availability, this could hold the sender of the availability, the intended recipient, the beneficial owner(s), the lender (which may differ from the sender as the lender may have the same piece of availability going through multiple agents), an agent or a venue.">
+    partyRole PartyRole (0..*) <"Defines the role(s) that party(ies) may have in relation to the inventory.">
+    availableInventoryRecord AvailableInventoryRecord (0..*) <"An array holding the list of inventory being described. Each element in the inventoryRecord array represents an individual piece of inventory i.e. a security.">
+	
+type AvailableInventoryRecord extends InventoryRecord: <"An individual piece of available inventory. This represents a single security and its associated criteria. The criteria are used to describe any restrictions on the securities.">
+    expirationDateTime zonedDateTime (0..1) <"There may be a set period/time restriction associated to the security.">
+    collateral CollateralProvisions (0..*) <"The type of collateral can often be required when determining if the piece of availability being described is suitable for a party.">
+    partyRole PartyRole (0..*) <"An individual security may be held by several agents. Including the party role at this level allows us to reference the party holding this specific item.">
+    priceQuantity PriceQuantity (0..*) <"Primarily used to hold the quantity of the security. An additional element can be used to hold an optional rate associated to this piece of availability.">
```

Comment:
Why `PriceQuantity` and not just `Quantity`, if it's only to hold the quantity?

And why `(0..*)`? If only 1 security, max cardinality should be 1. And should attribute really be optional?

Also I would have expected this quantity attribute to be in `InventoryRecord` rather than in the extended type. Does it make sense to talk about inventory without talking about a quantity?

Reply:
The ```PriceQuantity``` was used as we need to support the entry of a quantity and potentially a rate associated to the holding too. We also need to support the use case where a lender does not want to include either of these details, only the security code itself. This is why the cardinality was set to (0..*).
I’ve changed this to now use a ```Quantity``` type with cardinality (0..1) for the available quantity and a ```Price``` type with cardinality (0..1) for an optional rate against that quantity. I have also added a condition to confirm that if the price (rate) is present then it is set as an Interest rate in the priceType.